config/main.py

```diff
@@ -1313,5 +1315,274 @@ def naming_mode_alias():
+        config_db.mod_entry('SFLOW_SESSION', ifname, {'admin_state': 'up'})
+
+#
+# 'sflow' command ('config sflow interface disable  ...')
+#
+@interface.command()
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.pass_context
+def disable(ctx, ifname):
+    if not interface_name_is_valid(ifname) and ifname != 'all':
+        click.echo("Invalid interface name")
+        return
+
+    config_db = ctx.obj['db']
+    intf_dict = config_db.get_table('SFLOW_SESSION')
+
+    if intf_dict and ifname in intf_dict.keys():
+        intf_dict[ifname]['admin_state'] = 'down'
+        config_db.mod_entry('SFLOW_SESSION', ifname, intf_dict[ifname])
+    else:
+        config_db.mod_entry('SFLOW_SESSION', ifname,
+                            {'admin_state': 'down'})
+
+#
+# 'sflow' command ('config sflow interface sample-rate  ...')
+#
+@interface.command('sample-rate')
+@click.argument('ifname', metavar='<interface_name>', required=True, type=str)
+@click.argument('rate', metavar='<sample_rate>', required=True, type=int)
+@click.pass_context
+def sample_rate(ctx, ifname, rate):
+    if not interface_name_is_valid(ifname) and ifname != 'all':
+        click.echo('Invalid interface name')
+        return
+    if not is_valid_sample_rate(rate):
+        click.echo('Error: Sample rate must be between 256 and 8388608')
+        return
+
+    config_db = ctx.obj['db']
+    sess_dict = config_db.get_table('SFLOW_SESSION')
+
+    if sess_dict and ifname in sess_dict.keys():
+        sess_dict[ifname]['sample_rate'] = rate
+        config_db.mod_entry('SFLOW_SESSION', ifname, sess_dict[ifname])
+    else:
+        config_db.mod_entry('SFLOW_SESSION', ifname, {'sample_rate': rate})
+
+
+#
+# 'sflow collector' group
+#
+@sflow.group()
+@click.pass_context
+def collector(ctx):
+    """Add/Delete a sFlow collector"""
+    pass
+
+def is_valid_collector_info(name, ip, port):
+    if len(name) > 16:
+        click.echo("Collector name must not exceed 16 characters")
+        return False
+
+    if port not in range(0, 65535 + 1):
+        click.echo("Collector port number must be between 0 and 65535")
+        return False
+
+    if not is_ipaddress(ip):
+        click.echo("Invalid IP address")
+        return False
+
+    return True
+
+#
+# 'sflow' command ('config sflow collector add ...')
+#
+@collector.command()
+@click.option('--port', required=False, type=int, default=6343,
+              help='Collector port number')
+@click.argument('name', metavar='<collector_name>', required=True)
+@click.argument('ipaddr', metavar='<IPv4/v6_address>', required=True)
+@click.pass_context
+def add(ctx, name, ipaddr, port):
+    """Add a sFlow collector"""
+    ipaddr = ipaddr.lower()
+
+    if not is_valid_collector_info(name, ipaddr, port):
+        return
+
+    config_db = ctx.obj['db']
+    collector_tbl = config_db.get_table('SFLOW_COLLECTOR')
+
+    if (collector_tbl and name not in collector_tbl.keys() and len(collector_tbl) == 2):
```

Comment:
Collectors are limited to 2. Is this a limit from upper-layer hsflowd daemon?

Also confused about the output in the PR description that you were abled to configure 4 collectors.

Reply:
Because you were reading the output from an outdated version of the code. I changed that afterwards:

```
admin@sonic:~$
admin@sonic:~$ sudo config sflow collector add collect1 1.1.1.1
admin@sonic:~$ sudo config sflow collector add collect2 1.1.1.2 --port 6100
admin@sonic:~$ sudo config sflow collector add collect3 1.1.1.3 --port 6100
Only 2 collectors can be configured, please delete one
```

https://github.com/Azure/sonic-utilities/pull/592#commitcomment-34328258

Reply:
Is this a limitation from upper-layer hsflowd daemon?